etc/deploy/tracing/README.md

```diff
@@ -0,0 +1,57 @@
+# Overview
+
+Pachyderm has the ability to trace requests using Jaeger (an implementation of
+the OpenTracing API written by Uber). This can be useful when diagnosing slow
+clusters.
+
+TODO add screenshot
+
+# Quickstart
+
+## Collecting Traces
+
+To use tracing in Pachyderm, you need to:
+1. Run Jaeger in Kubernetes
+
+    ```
+    kubectl apply -f etc/deploy/tracing/jaeger-all-in-one.yaml
+    ```
+
+2. Point Pachyderm at Jaeger
+
+    ```
+    # For pachctl
+    $ export JAEGER_ENDPOINT=localhost:14268
+    $ jaeger_pod="$(kubectl get po -l app=jaeger -o jsonpath='{.items[].metadata.name}')"
+    $ kubectl port-forward "po/${jaeger_pod}" 14268 &  # Collector port
+
+    # For pachd
+    $ kubectl delete po -l suite=pachyderm,app=pachd
+    ```
+    The port-forward command is necessary because `pachctl` sends traces to
+    Jaeger (it actually initiates every trace), and reads the `JAEGER_ENDPOINT`
+    environment variable for the address to which it will send the trace info.
+
+    Restarting the pachd pod is necessary because pachd also sends trace
+    information to Jaeger, but it reads the environment variables corresponding
+    to the Jaeger service[1] on startup to find Jaeger (the Jaeger service is
+    created by the `jaeger-all-in-one.yaml` manifest). Killing the pods
+    restarts them, which causes them to connect to Jaeger.
+
+3. Send Pachyderm a traced request
+
+    Just set the `--trace` flag with any `pachctl` command:
+    ```
+    pachctl --trace list-job # for example
+    ```
+
+## Viewing Traces
+Just run:
+```
+# See step 2 under "Collecting Traces" above for 'jaeger_pod' definition
+$ kubectl port-forward "po/${jaeger_pod}" 16686 &  # UI port
+```
+then connect to `localhost:16686` in your browser and you should see all
+collected traces.
+
+[1] https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables
```

Comment:
Good writeup, can we relocate it to `doc/deployment`, `etc` is a good place for the `yaml` file but not docs. We'll probably want to point users at this sometime soon and having it rendered on RTD will be nice for that.

Reply:
Done